### ISSUE TITLE
spliced image forgery localization

### ISSUE THREAD

**Issue by bareqraad** [brahmadaityo/image_splicing-localization#1]
am trying to work on autoencoder to localize the forged region in the image I fed the autoencoder with the tampered image after converting it to YCbCr then isolate the CB band to the encoder as input the output on the other hand is the ground truth of that input hopefully the model learns the features help him to isolate future samples I didn't get good results though any help with the network or the methodology 
`from tensorflow.keras.layers import Conv2D, UpSampling2D
from tensorflow.keras.models import Sequential
import numpy as np
import matplotlib.pyplot as plt
import os
import cv2

def load_images_from_folder(TPfolder,GTfolder):
    TP_images = []    
    GT_images = []    
    for filename in os.listdir(TPfolder):
        try:
            file_tp = os.path.join(TPfolder,filename)
            #file_gt =  os.path.join(GTfolder,os.path.splitext(os.path.basename(file_tp))[0] + '.png')
            file_gt =  os.path.join(GTfolder,os.path.splitext(os.path.basename(file_tp))[0] + '_gt.png')
            img1 = cv2.imread(file_tp,cv2.COLOR_BGRYCRCB).astype(np.float32)/255.0
            img1 = cv2.resize(img1,(256,256))
            img2 =cv2.imread(file_gt,0).astype(np.float32)/255.0
            img2 = cv2.resize(img2,(256,256))
            if img1 is not None and img2 is not None:                
                TP_images.append(np.array(img1[:,:,1]))
                GT_images.append(np.array(img2))
                print(np.array(img1).shape)
        except:
            True

    return TP_images,GT_images
TPfolder = 'D:/casia2/'
GTfolder = 'D:/casia2mask/'
TP,GT = np.array(load_images_from_folder(TPfolder,GTfolder))

from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split( TP, GT, train_size=0.8, random_state=42)

x_train = np.reshape(x_train, (len(x_train), 256,256,1))
y_train = np.reshape(y_train, (len(y_train), 256,256,1))
x_test = np.reshape(x_test, (len(x_test), 256,256,1))
y_test = np.reshape(y_test, (len(y_test), 256,256,1))

TP = []
GT = []

#Encoder

model = Sequential()
model.add(Conv2D(64, (3, 3), activation='relu', padding='same', strides=2, input_shape=(256, 256, 1)))
model.add(Conv2D(128, (3, 3), activation='relu', padding='same'))
model.add(Conv2D(128, (3,3), activation='relu', padding='same', strides=2))
model.add(Conv2D(256, (3,3), activation='relu', padding='same'))
model.add(Conv2D(256, (3,3), activation='relu', padding='same', strides=2))
model.add(Conv2D(512, (3,3), activation='relu', padding='same'))
model.add(Conv2D(512, (3,3), activation='relu', padding='same'))
model.add(Conv2D(256, (3,3), activation='relu', padding='same'))

model.add(Conv2D(128, (3,3), activation='relu', padding='same'))
model.add(UpSampling2D((2, 2)))
model.add(Conv2D(64, (3,3), activation='relu', padding='same'))
model.add(UpSampling2D((2, 2)))
model.add(Conv2D(32, (3,3), activation='relu', padding='same'))
model.add(Conv2D(16, (3,3), activation='relu', padding='same'))
model.add(Conv2D(1, (3, 3), activation='relu', padding='same'))
model.add(UpSampling2D((2, 2)))
model.compile(optimizer='adam',loss='mse' , metrics=['accuracy'])
model.summary()

model.fit(x_train,y_train, epochs=100, batch_size=50)

###########################################################
model.save('f:/a.model')
model.evaluate(x_test, y_test)

localise = model.predict(x_test)

`
